iot-management-device/src/main/kotlin/com/example/iotmanagementdevice/mapper/DeviceNotificationMapper.kt

```diff
@@ -0,0 +1,24 @@
+package com.example.iotmanagementdevice.mapper
+
+import com.example.internal.commonmodels.DeviceUpdateNotification
+import com.example.internal.input.reqreply.device.update.proto.UpdateDeviceResponse
+import com.google.protobuf.Timestamp
+import org.springframework.stereotype.Component
+import java.time.Instant
+
+@Component
+class DeviceNotificationMapper {
+    fun toDeviceUpdateNotification(response: UpdateDeviceResponse): DeviceUpdateNotification {
+        val currentTimestamp = Instant.now().let {
+            Timestamp.newBuilder().apply {
+                seconds = it.epochSecond
+                nanos = it.nano
+            }.build()
```

Comment:
what if event is consumed with a big delay from time when UpdateDeviceResponse is published to kafka?